src/renderers/testing/ReactTestRenderer.js

```diff
@@ -60,13 +62,15 @@ ReactTestComponent.prototype.receiveComponent = function(
   this._currentElement = nextElement;
   this.updateChildren(nextElement.props.children, transaction, context);
 };
+
 ReactTestComponent.prototype.getHostNode = function() {};
-ReactTestComponent.prototype.getPublicInstance = function() {
-  // I can't say this makes a ton of sense but it seems better than throwing.
-  // Maybe we'll revise later if someone has a good use case.
-  return null;
+
+ReactTestComponent.prototype.getPublicInstance = function(transaction) {
+  return transaction.getMockConfig().getMockRef(this._currentElement);
```

Comment:
Let's make this three lines. First get the element, then get the options, then use them to get and return the ref.